internal/rbd/rbd_util.go

```diff
@@ -979,11 +921,36 @@ func (rv *rbdVolume) getImageInfo() error {
 		return err
 	}
 	rv.imageFeatureSet = librbd.FeatureSet(features)
-	err = rv.updateVolWithImageInfo(rv.conn.Creds)
+
+	// Get parent information.
+	// TODO: Replace GetParentInfo() after
+	// https://github.com/ceph/go-ceph/issues/347 is fixed.
+	parentPool := make([]byte, 128)
+	parentName := make([]byte, 128)
+	parentSnapname := make([]byte, 128)
+	err = image.GetParentInfo(parentPool, parentName, parentSnapname)
+	if err != nil {
+		// Caller should decide whether not finding
+		// the parent is an error or not.
+		if strings.Contains(err.Error(), "No such file or directory") {
```

Comment:
go-ceph should return a proper error message which we can use for error comparison instead of string comparison

example
```
if errors.Is(err, librbd.ErrNotFound) {
```

Reply:
We don't have that in current implementation of go-ceph, probably we can do that in https://github.com/ceph/go-ceph/issues/347 and keep this PR on hold till then.

@nixpanic @phlogistonjohn

Reply:
The error string would currently by `rbd: ret=2, No such file or directory`.

`GetParentInfo()` does not call `getError()`, so it will not return `librbd.ErrNotFound`. The errors in `rbd/errors.go` from go-ceph should probably be converted to instances of `rbdError` so that they can match the errors returned by `rbdError()` or `getError()`.

Reply:
yes as it won't affect any functionality. in that case we can take this in and make the error handling changes when new go-ceph release is available

Reply:
I read over this thread a few times and I'm not entirely sure what the issue is still. ErrNotFound is not meant to be equivalent to ENOENT (although maybe it should be called ErrImageNotFound). Regardless CSI can still match on errno value rather than string. We recently discussed this in go-ceph on [issue 343](github.com/ceph/go-ceph/issues/343)

Making the existing errors in rbd/errors.go based on rbdError but with extra context is a good idea but I'm not sure it would help here. Let's discuss that in an issue over at go-ceph.

Reply:
Yes, CSI can match on error value rather than string.

The issue here is, because GetParentInfo() currently doesn't call getError() it won't be returning librbd.ErrNotFound else it would have returned that in case of ENOENT and it would have been easier/cleaner to compare with that.

Right now, comparing with actual error value or string are both same.